packages/opentelemetry-scope-zone-peer-dep/src/ZoneScopeManager.ts

```diff
@@ -0,0 +1,258 @@
+ * It heavily depends on [zone.js]{@link https://www.npmjs.com/package/zone.js}.
+ * It stores the information about scope in zone. Each Scope will have always new Zone;
+ * It also supports binding a certain Span to a target that has "addEventListener" and "removeEventListener".
+ * When this happens a new zone is being created and the provided Span is being assigned to this zone.
+ */
+export class ZoneScopeManager implements ScopeManager {
+  /**
+   * whether the scope manager is enabled or not
+   */
+  private _enabled = false;
+
+  /**
+   * Helps to create a unique name for the zones - part of zone name
+   */
+  private _zoneCounter = 0;
+
+  /**
+   * Returns the active scope from certain zone name
+   * @param activeZone
+   */
+  private _activeScopeFromZone(
+    activeZone: Zone | undefined
+  ): unknown | undefined {
+    return activeZone && activeZone.get(ZONE_SCOPE_KEY);
+  }
+
+  /**
+   * @param target Function to be executed within the scope
+   * @param scope A scope (span) to be executed within target function
+   */
+  private _bindFunction<T extends Function>(target: T, scope?: unknown): T {
+    const manager = this;
+    const contextWrapper = function(this: any, ...args: unknown[]) {
+      return manager.with(scope, () => target.apply(this || scope, args));
+    };
+    Object.defineProperty(contextWrapper, 'length', {
+      enumerable: false,
+      configurable: true,
+      writable: false,
+      value: target.length,
+    });
+    return (contextWrapper as unknown) as T;
+  }
+
+  /**
+   * @param obj target object on which the listeners will be patched
+   * @param scope A scope (span) to be bind to target
+   */
+  private _bindListener<T>(obj: T, scope?: unknown): T {
+    const target = (obj as unknown) as TargetWithEvents;
+    if (target.__ot_listeners !== undefined) {
+      return obj;
+    }
+    target.__ot_listeners = {};
+
+    if (typeof target.addEventListener === 'function') {
+      target.addEventListener = this._patchAddEventListener(
+        target,
+        target.addEventListener,
+        scope
+      );
+    }
+
+    if (typeof target.removeEventListener === 'function') {
+      target.removeEventListener = this._patchRemoveEventListener(
+        target,
+        target.removeEventListener
+      );
+    }
+
+    return obj;
+  }
+
+  /**
+   * Creates a new unique zone name
+   */
+  private _createZoneName() {
+    this._zoneCounter++;
+    const random = Math.random();
+    return `${this._zoneCounter}-${random}`;
+  }
+
+  /**
+   * Creates a new zone
+   * @param zoneName zone name
+   * @param scope A scope (span) to be bind with Zone
+   */
+  private _createZone(zoneName: string, scope: unknown): Zone {
+    return Zone.root.fork({
+      name: zoneName,
+      properties: {
+        [ZONE_SCOPE_KEY]: scope,
```

Comment:
What's the role of the `[` and `]` here? Is that different from `properties: { ZONE_SCOPE_KEY: scope }`? (Maybe this is a JS syntax thing I haven't seen yet)

Reply:
this is how you setup the key of object in js using the variable value.